ocaml/lib/snarky_bindings.ml

```diff
@@ -113,43 +115,218 @@ module Field' = struct
   let to_constant_and_terms x = Field.to_constant_and_terms x
 end
 
+let add_gate (label : string) gate =
+  Impl.with_label label (fun () ->
+      Impl.assert_
+        { annotation = None
+        ; basic =
+            Kimchi_backend_common.Plonk_constraint_system.Plonk_constraint.T
+              gate
+        } )
+
 module Gates = struct
+  let zero in1 in2 out =
+    add_gate "zero"
+      (Raw { kind = Zero; values = [| in1; in2; out |]; coeffs = [||] })
+
   let generic sl l sr r so o sm sc =
-    Impl.with_label "generic_gate" (fun () ->
-        Impl.assert_
-          { annotation = Some __LOC__
-          ; basic =
-              Kimchi_backend_common.Plonk_constraint_system.Plonk_constraint.T
-                (Basic { l = (sl, l); r = (sr, r); o = (so, o); m = sm; c = sc })
-          } )
+    add_gate "generic_gate"
```

Comment:
nit: I think `generic` is good enough in this context, no need to explicitly specify `_gate`